app/src/main/java/org/openobservatory/ooniprobe/activity/TextActivity.java

```diff
@@ -39,11 +93,83 @@ public static Intent newIntent(Context context, String text) {
 	@Override public boolean onOptionsItemSelected(MenuItem item) {
 		switch (item.getItemId()) {
 			case R.id.clipboard:
-				((ClipboardManager) getSystemService(CLIPBOARD_SERVICE)).setPrimaryClip(ClipData.newPlainText(getString(R.string.General_AppName), textView.getText().toString()));
+				((ClipboardManager) getSystemService(CLIPBOARD_SERVICE)).setPrimaryClip(ClipData.newPlainText(getString(R.string.General_AppName), text));
 				Toast.makeText(this, R.string.Toast_CopiedToClipboard, Toast.LENGTH_SHORT).show();
 				return true;
 			default:
 				return super.onOptionsItemSelected(item);
 		}
 	}
+
+	public void showText(){
+		switch (getIntent().getIntExtra(TYPE, 0)) {
+			case TYPE_JSON:
+				//Try to open file, if it doesn't exist dont show Error dialog immediately but try to download the json from internet
+				try {
+					File entryFile = Measurement.getEntryFile(this, measurement.id, measurement.test_name);
+					String json = FileUtils.readFileToString(entryFile, Charset.forName("UTF-8"));
+					json = new GsonBuilder().setPrettyPrinting().disableHtmlEscaping().create().toJson(new JsonParser().parse(json));
+					text = json;
+					webView.loadDataWithBaseURL(null, json, "text/plain", "UTF-8", null);
+					//webView.loadData(json, "text/plain", "UTF-8");
+				} catch (Exception e) {
+					e.printStackTrace();
+					if (ReachabilityManager.getNetworkType(this).equals(ReachabilityManager.NO_INTERNET)) {
+						new MessageDialogFragment.Builder()
+								.withTitle(getString(R.string.Modal_Error))
+								.withMessage(getString(R.string.Modal_Error_RawDataNoInternet))
+								.build().show(getSupportFragmentManager(), null);
```

Comment:
It looks like you need to `return` here?

Reply:
done